project/ianlienfa/README.rst

```diff
@@ -0,0 +1,122 @@
+  When researching on dominance relationships or problem-specific heuristics on OR problems, 
+  We often have to first implement the best heuristic and pruning strategy known to us.
+  Then we can test out different search strategies and branch and pruning strategies.
+  This framework is designed to be elastic enough to be able to easily insert and remove strategies
+  
+2. Provide an simple api for RL on branch and bound problem solving (*Main Goal of this Project*)
+
+Prospective Users
+=================
+
+1. Researchers that are interested in the framework and want to try out different search, branch and pruning strategies for OR problems.
+2. Students who would like to solve problem using branch and bound
+3. Industry professionals that would like to try out some ideas before moving onto implementing an state-of-the-art solver
+
+System Architecture
+===================
+
+`BB_engine`: 
+  This is the main engine of the framework.  It is responsible for the following:
+  1. Read the input file and parse the input
+  2. Construct the environment
+  3. Run the search strategy
+  4. Output the results
+
+`branch_modules`:
+  This is the collection of branch classes
+  A branch class should inherit the base class `BranchMod`, which requires the following:
+  * Implement the `branch` function
+
+`prune_modules`:
+  This is the collection of pruning classes
+  A pruning class should inherit the base class `PruneMod`, which requires the following:
+  * Implement the `prune` function
+
+`Node`:
+  This is the collection of node classes
+  A specific problem requires a node class and a graph class to be defined.
+  It generically specifies the node information during branch and bound
+
+`Graph`:
+  The graph class is used to store the graph information during branch and bound
+  Compared to the node class, the graph class has more global information and knows about the state of the search
+  For example, a graph class for the :math: `1|\Sigma_Cj` problem, we need:
+    * the configuration of the probelm instance
+    * the best value founded so far    
+    * the data structure for searching
+
+`config.h`:
+  The configuration for the problem solving
+  Users can twiddle with the configuration in this file, 
+  including the problem type, the searching strategy, the branching strategy, and the pruning strategy
+
+In general, the framework calls the BB_engine at the start of the program, 
+then for every node visited, the framework calls the branch_modules to branch, 
+the prune_modules to prune, updates the graph, then relies on the search_module to search for the next node.
+
+The users can use this system solely from source, twiddle the configuration in the config.h file, and build them from the source.  
+For this project, we hope to wrap it in a python package, and provide a gym-like api for RL on branch and bound problem solving.
```

Comment:
Which modules will be in Python?  If all modules are in C++, what parts of the modules are planned to be exposed to Python for controlling?  What flexibility in control you plan to offer?

Reply:
A more specific python module descriptions added, thanks!

Reply:
LGTM.